### PR TITLE
fix(test): #228 Start memcached as part of 'npm test'

### PR DESCRIPTION
**Pull request by deeptibaghel**
Have added the lines to start memcached, not checking if it is running already as it does nothing if it is already running. spawnSync makes it waiting forever so i am using spawn only.

### CODE REVIEW

**Comment by deeptibaghel** [comment]
@vladikoff i am not clear on which test has timed out in travis

**Comment by vladikoff** [comment]
hm in the end it shows: "test/local/ban_tests.js timeout!:
     timeout!
  "

File: test/memcache-helper.js [hunk 1]
```diff
@@ -37,6 +36,9 @@ var config = {
   }
 }
 
+const spawn = require('child_process').spawn
+spawn('memcached', ['-v'])
```
Comment (vladikoff) [inline test/memcache-helper.js]: try using spawnSync? it might now spawn fast enough for tests asynchronously

File: test/memcache-helper.js [hunk 2]
```diff
@@ -10,8 +10,7 @@ P.promisifyAll(Memcached.prototype)
 
 var config = {
   memcache: {
-    host: '127.0.0.1',
-    port: '11211'
+    address: '127.0.0.1:11211',
```
Comment (vladikoff) [inline test/memcache-helper.js]: what does this changed do?

Comment (deeptibaghel) [reply test/memcache-helper.js]: 'address' is being used here but never defined :
https://github.com/mozilla/fxa-customs-server/blob/be3b5794da2df8d8d7d042a269ef3ca47e248c1f/test/memcache-helper.js#L41

**Comment by vladikoff** [comment]
@deeptibaghel I also get this locally:

```
test/local/ban_tests.js ............................... 0/1 31s
  not ok timeout
```

checking now....

File: test/memcache-helper.js [hunk 3]
```diff
@@ -37,6 +36,9 @@ var config = {
   }
 }
 
+const spawn = require('child_process').spawnSync
```
Comment (vladikoff) [inline test/memcache-helper.js]: @deeptibaghel ah I see so we would need to use `spawn` here. I Think the tests hang because the process is left running after the tests are done in `ban_tests.js`

Comment (vladikoff) [reply test/memcache-helper.js]: `spawnSync` just blocks the process and doesn't continue because the `memcached` just sits there

**Comment by deeptibaghel** [comment]
@vladikoff but travis failure is not related to spawn i think. I added those 2 parameters but timeout still persists.

**Comment by vladikoff** [comment]
@deeptibaghel Let's make a change in the README.md: 

https://github.com/mozilla/fxa-customs-server/blob/master/README.md#testing we can remove the 

> On Mac OS X, memcached must be manually started for the tests to run. 

section. 

Add a section into https://github.com/mozilla/fxa-customs-server/blob/master/README.md#development to mention that `memcached` needs to be installed

**Comment by vladikoff** [comment]
We already have:

> You'll also need to install memcached, otherwise all requests will be blocked 

but we should move that around a bit, it is mentioned a bit late in the README.

**Comment by vladikoff** [comment]
@deeptibaghel are you getting timeouts locally running:

```
 node_modules/.bin/tap -R spec test/local test/remote --coverage --cov
```

**Comment by deeptibaghel** [comment]
> @deeptibaghel are you getting timeouts locally running:
> 
>  node_modules/.bin/tap -R spec test/local test/remote --coverage --cov

I am not getting any error for test/local/ban_tests.js.

**Comment by deeptibaghel** [comment]
@vladikoff i have modified README.md

File: test/memcache-helper.js [hunk 4]
```diff
@@ -37,6 +36,12 @@ var config = {
   }
 }
 
+const spawn = require('child_process').spawn
```
Comment (vladikoff) [inline test/memcache-helper.js]: @deeptibaghel I talked to @vbudhram about this and we suspect `spawn` starts the process, but tests timeout because it is not cleaned up after the last test. We could write a function to stop the process, but we can just a simpler solution for now: Start memcached in this file https://github.com/mozilla/fxa-customs-server/blob/master/scripts/test-local.sh and run `kill` on it after tests are done. Edge case to test is the case where it is already running, see what happens.

Example: https://github.com/mozilla/fxa-auth-server/blob/master/scripts/start-local-mysql.sh#L8-L17

**Comment by deeptibaghel** [comment]
@vladikoff ,  i think it's now working the way we wanted, the test script first checks if memcached is running, starts if not and kills later.

**Comment by vladikoff** [comment]
@deeptibaghel Thank you!! 👍
